rust/core-lib/src/styles.rs

```diff
@@ -220,4 +222,65 @@ impl ThemeStyleMap {
         self.styles.push(style.clone());
         result
     }
-}
+
+    /// Delete key from the themes map.
+    pub(crate) fn remove_theme(&mut self, path: &Path) -> Option<String> {
+        if let Some(theme_name) = path.file_stem().and_then(OsStr::to_str) {
+            self.themes.themes.remove(theme_name);
+            //Delete dump file
+            let _ = fs::remove_file(path.with_extension("tmdump"));
+            Some(theme_name.to_string())
+        } else { None }
+    }
+
+    /// Load all themes inside the given directory.
+    pub(crate) fn load_theme_dir(&mut self, themes_dir: PathBuf) {
+        match ThemeSet::discover_theme_paths(themes_dir) {
+            Ok(themes) => {
+                for theme in themes.iter() {
+                    self.load_theme(theme);
+                }
+            },
+            Err(e) => eprintln!("Error while loading themes dir: {:?}",e),
+        }
+    }
+
+    fn load_theme(&mut self, path: &Path) {
+        //Check for the dump file first
+        let theme_cache = path.with_extension("tmdump");
+        if theme_cache.exists() {
+            match from_dump_file(theme_cache) {
+                Ok(theme) => self.update_theme_map(path, theme),
+                Err(_) => self.insert_theme(path),
+            };
+        } else {
+            self.insert_theme(path);
+        }
+    }
+
+    /// Insert theme into the map.
+    /// Store binary dump in a file with `tmdump` extension.
+    pub(crate) fn insert_theme(&mut self, path: &Path) 
+        -> Option<String>
+    {
+        match ThemeSet::get_theme(path) {
+            Ok(theme) => {
+                dump_to_file(&theme, path.with_extension("tmdump")).unwrap();
```

Comment:
I'd prefer to keep caches somewhere that isn't so user facing. A simple solution for now would be to just use $theme_dir/cache? At some point we might want the client to specify where caches should be written.